### ISSUE TITLE
Php error in console command returns wrong exit code

### ISSUE THREAD

**Issue by b1rdex** [symfony/symfony#54351]
### Symfony version(s) affected

v7.0.4

### Description

Error in PHP code throws an Error. This error is rendered with error style. But the command exit code is 0.

### How to reproduce

```php
#[AsCommand(
    name: 'app:error',
)]
class ErrorCommand extends Command
{
    protected function execute(InputInterface $input, OutputInterface $output): int
    {
        $foo = null;
		// next line throws an Error
        $foo->bar();

        return Command::INVALID;
	}
}
```

Run `symfony console app:error ; echo 'exit code: '$?`

See the `0` exit code:

```
❯ symfony console app:error ; echo $?

In ErrorCommand.php line 27:
                                           
  Call to a member function bar() on null  
                                           

app:test-import

exit code: 0
```

### Possible Solution

Here was a wrong idea, I removed it to simplify the issue. See the comments.

**Comment by b1rdex** [comment]
Looks like the possible solution debug information is wrong. I checked `ConsoleErrorEvent::getExitCode` and it never returns 0. So the problem is somewhere else.

**Comment by b1rdex** [comment]
Ok, got it. If I set `\Symfony\Component\Console\Application::$catchErrors = true` then I got a correct exit code = 1.

Is this by design? Why default runtime (`vendor/autoload_runtime.php` + `SymfonyRuntime`) hides the command execution error with exit code 0?

**Comment by xabbuh** [comment]
Same as #53946?

**Comment by b1rdex** [comment]
Yes it is. Thanks. I fixed it the same way as in https://github.com/symfony/symfony/issues/53946#issuecomment-1954789389.

**Comment by stof** [comment]
Note that the root cause of the issue should be fixed in the next PHP 8.3.x release as a fix has been merged in php-src.
